File: packages/algo-py/solutions/algo_02_test.py
import unittest
from solutions import TreeNode, is_valid_BST


class TestAlgo_01(unittest.TestCase):

    def test_valid(self):
        root = TreeNode(5)
        root.left = TreeNode(4)
        root.right = TreeNode(6)
        self.assertTrue(is_valid_BST(root, None, None))

    def test_invalid(self):
        root = TreeNode(5)
        root.left = TreeNode(6)
        root.right = TreeNode(4)
        self.assertFalse(is_valid_BST(root, None, None))


if __name__ == '__main__':
    unittest.main()

File: packages/algo-go/solutions/algo_02_test.go
package solutions

import "testing"

func TestIsValid(t *testing.T) {
	root := TreeNode{
		val:   5,
		left:  NewTreeNode(4),
		right: NewTreeNode(6),
	}
	if !IsValidBST(&root, nil, nil) {
		t.Error("Failed")
	}
}

func TestIsInvalid(t *testing.T) {
	root := TreeNode{
		val:   5,
		left:  NewTreeNode(7),
		right: NewTreeNode(6),
	}
	if IsValidBST(&root, nil, nil) {
		t.Error("Failed")
	}
}
